src/mimeparser.rs

```diff
@@ -1233,6 +1234,7 @@ impl MimeMessage {
                 return Ok(());
             }
         }
+        let mut param = Params::new();
```

Comment:
Nit, feel free to ignore: With this chnange, if someone should write `part.param.set(...)` instead of `param.set(...)`, then the param wouldn't be changed at all because it will later be overwritten by the line `part.param = param;`

I would very slightly prefer to write `let mut part = Part::default();` up here already, and to write `part.param.set(...)` throughout the function, in order to avoid having this duplicate `Param` (one in the `param` variable and one inside the `part`).